src/main/scala/org/clulab/wm/eidos/groundings/grounders/SRLCompositionalGrounder.scala

```diff
@@ -357,14 +429,15 @@ class SRLCompositionalGrounder(name: String, domainOntology: DomainOntology, w2v
 
   private def groundToBranches(branches: Seq[String], span: Interval, s: Sentence, topN: Option[Int], threshold: Option[Float]): OntologyGrounding = {
     val patterns = branches.flatMap(conceptPatternsSeq(_))
+    val examples = branches.flatMap(conceptExamplesSeq(_))
     val embeddings = branches.flatMap(conceptEmbeddingsSeq(_))
```

Comment:
This can be definitely be rearranged elsewhere if that makes things easier.

Reply:
It seems like rather than combining the many nodes of the multiple ontology branches first, one could ground against the individual branches and then combine those few (at least if there is a topN) results.  Just now it looks like
```scala
  private def groundToBranches(branches: Seq[String], span: Interval, s: Sentence, topN: Option[Int], threshold: Option[Float]): OntologyGrounding = {
    val contentWords = canonicalizer.canonicalWordsFromSentence(s, span).toArray
    val branchesGroundings = branches.flatMap { branch =>
      val patterns = conceptPatternsMap(branch)
      val examples = conceptExamplesMap(branch)
      val embeddings = conceptEmbeddingsMap(branch)
      val branchGroundings = groundPatternsThenEmbeddings(contentWords, patterns, examples, embeddings)

      // Do this first for each branch and then
      filterAndSlice(branchGroundings, topN, threshold)
    }
    // at the end, filter the combined values if appropriate.
    val filtered =
        if (branches.length == 1) branchesGroundings
        else filterAndSlice(branchesGroundings, topN, threshold)

    newOntologyGrounding(filtered)
  }
```

Reply:
That makes sense!